lib/modules/forgot_password/forgot_password_view.dart

```diff
@@ -0,0 +1,208 @@
+part of 'forgot_password_module.dart';
+
+abstract class ForgotPasswordView extends View<ForgotPasswordViewDelegate>
+    with ProgressHUDViewMixin, AlertViewMixin {
+  static const showingResetSuccessfullyNotificationDuration =
+      Duration(milliseconds: 600);
+  static const dismissResetSuccessfullyNotificationDelayDuration =
+      Duration(seconds: 2);
+
+  static const emailTextFieldKey = Key("email_text_field_key");
+  static const resetButtonKey = Key("reset_button_key");
+  static const resetSuccessfullyNotificationSlideTransitionKey =
+      Key("reset_successfully_notification_slide_transition");
+  static const resetSuccessfullyNotificationKey =
+      Key("reset_successfully_notification");
+
+  static const resetSuccessfullyNotificationSlideTransitionBeginOffset =
+      Offset(0, -1);
+  static const resetSuccessfullyNotificationSlideTransitionEndOffset =
+      Offset.zero;
+
+  void clearEmailTextField();
+
+  void setResetButton({required bool isEnabled});
+
+  void showResetSuccessfullyNotification();
+}
+
+abstract class ForgotPasswordViewDelegate implements AlertViewMixinDelegate {
+  BehaviorSubject<String> get emailTextFieldDidChange;
+
+  BehaviorSubject<String> get resetButtonDidTap;
+}
+
+class ForgotPasswordViewImpl extends StatefulWidget {
+  const ForgotPasswordViewImpl({Key? key}) : super(key: key);
+
+  @override
+  _ForgotPasswordViewImplState createState() => _ForgotPasswordViewImplState();
+}
+
+class _ForgotPasswordViewImplState extends ViewState<ForgotPasswordViewImpl,
+        ForgotPasswordModule, ForgotPasswordViewDelegate>
+    with
+        ProgressHUDViewMixin<ForgotPasswordViewDelegate>,
+        SingleTickerProviderStateMixin,
+        AlertViewMixin
+    implements ForgotPasswordView {
+  late final AnimationController _notificationAnimationController =
+      AnimationController(
+    duration: ForgotPasswordView.showingResetSuccessfullyNotificationDuration,
+    vsync: this,
+  );
+
+  late final Animation<Offset> _notificationAnimation = Tween<Offset>(
+    begin: ForgotPasswordView
+        .resetSuccessfullyNotificationSlideTransitionBeginOffset,
+    end: ForgotPasswordView
+        .resetSuccessfullyNotificationSlideTransitionEndOffset,
+  ).animate(_notificationAnimationController);
+
+  final _emailTextController = TextEditingController();
+  final _isResetButtonEnabled = BehaviorSubject<bool>.seeded(false);
```

Comment:
it seems a bit strange for me to see this in `view` 🤔  shouldn't the state be kept & process in `presenter`?

Reply:
@sleepylee  It likes we both have view and view model in the same file 😁

Reply:
you can review the idea again later, technically, if you want to make an `Observable` here, maybe you should have changed the function:
`void setResetButton({required bool isEnabled});` into returning an `Observable`? 

In this case, I support you in making the state `_isResetButtonEnabled` reactive. But this approach is a bit cumbersome right now. It let the Presenter make the call `setResetButton(bool)`, then it applies to the local `BehaviorSubject`. 
So how about we keep all the BehaviorSubject in Presenter, including this, expose read-only Observable to here, then you can map it to your enable state handle of this button? Lol in this case you make a function but we recommend the opposite side 😄  you can refactor it later, but let me know if you get the idea or having objection

Reply:
@sleepylee I gonna document what I'm doing in this project also the reason why I'm doing this, so meet you there to solve your concern 😁